### PR TITLE
Local Variables

### PR DESCRIPTION
**Pull request by serkm**
Allocate local variables directly on the stack, with a special stack frame

### CODE REVIEW

File: libraries/llvm/rts.ll [hunk 1]
```diff
@@ -308,6 +308,57 @@ define ptr @getPointer(%Reference %reference, i64 %index, %Stack %stack) {
...
+    %base = load %Base, ptr %basePointer
+    %varPointer = getelementptr i8, %Base %base, i64 %offset
+    ret ptr %varPointer
+}
+
+define %Reference @newReference(%Stack %stack) alwaysinline {
+    %stackPointerPointer = getelementptr %StackValue, %Stack %stack, i64 0, i32 1, i32 0
+    %basePointer = getelementptr %StackValue, %Stack %stack, i64 0, i32 1, i32 1
+
+    %stackPointer = load %StackPointer, ptr %stackPointerPointer
```
Comment (b-studios) [inline libraries/llvm/rts.ll:344]: Nitpick, we now try to use the convention `Foo_pointer`. So this will be `stackPointer_pointer`

**Comment by b-studios** [comment]
Great job! Do you have results on running benchmarks on this?

**Comment by phischu** [comment]
Could you rebase onto current master? I can not reproduce the bad benchmarks, we should discuss offline.

**Review by b-studios — APPROVED** [review]
LGTM :)

**Pull request merged**